scylla/src/transport/downgrading_consistency_retry_policy.rs

```diff
@@ -0,0 +1,625 @@
+use scylla_cql::{
+    errors::{DbError, QueryError, WriteType},
+    frame::types::LegacyConsistency,
+    Consistency,
+};
+
+use crate::retry_policy::{QueryInfo, RetryDecision, RetryPolicy, RetrySession};
+
+/// Downgrading consistency retry policy - retries with lower consistency level if it knows\
+/// that the initial CL is unreachable. Also, it behaves as [DefaultRetryPolicy] when it believes that the initial CL is reachable.
+/// Behaviour based on [DataStax Java Driver]\
+///(https://docs.datastax.com/en/drivers/java/3.11/com/datastax/driver/core/policies/DowngradingConsistencyRetryPolicy.html)
+#[derive(Debug)]
+pub struct DowngradingConsistencyRetryPolicy;
+
+impl DowngradingConsistencyRetryPolicy {
+    pub fn new() -> DowngradingConsistencyRetryPolicy {
+        DowngradingConsistencyRetryPolicy
+    }
+}
+
+impl Default for DowngradingConsistencyRetryPolicy {
+    fn default() -> DowngradingConsistencyRetryPolicy {
+        DowngradingConsistencyRetryPolicy::new()
+    }
+}
+
+impl RetryPolicy for DowngradingConsistencyRetryPolicy {
+    fn new_session(&self) -> Box<dyn RetrySession> {
+        Box::new(DowngradingConsistencyRetrySession::new())
+    }
+
+    fn clone_boxed(&self) -> Box<dyn RetryPolicy> {
+        Box::new(DowngradingConsistencyRetryPolicy)
+    }
+}
+
+pub struct DowngradingConsistencyRetrySession {
+    was_retry: bool,
+}
+
+impl DowngradingConsistencyRetrySession {
+    pub fn new() -> DowngradingConsistencyRetrySession {
+        DowngradingConsistencyRetrySession { was_retry: false }
+    }
+}
+
+impl Default for DowngradingConsistencyRetrySession {
+    fn default() -> DowngradingConsistencyRetrySession {
+        DowngradingConsistencyRetrySession::new()
+    }
+}
+
+impl RetrySession for DowngradingConsistencyRetrySession {
+    fn decide_should_retry(&mut self, query_info: QueryInfo) -> RetryDecision {
```

Comment:
let's sprinkle a few logs (`debug` level should be fine) here, to be able to see that a consistency downgrade was performed

Reply:
How verbose should the logs be? Should they only inform about a consistency downgrade being performed or also contain node information etc.?

Reply:
info about consistency downgrade can be printed to `debug`, and additional spammy information (like nodes) can be either ignored or printed in `trace` level